sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CsvOutputWriter.scala

```diff
@@ -39,6 +39,10 @@ class CsvOutputWriter(
 
   private val gen = new UnivocityGenerator(dataSchema, writer, params)
 
+  if (params.bom) {
+    writer.write(0xFEFF)
```

Comment:
More details: https://stackoverflow.com/questions/32072017/write-utf-8-bom-with-supercsv

Reply:
Hm, `0xFEFF` is the BOM for UTF-16 Big Endian, see https://en.wikipedia.org/wiki/Byte_order_mark. Does it work if you specify `0xEFBBBF` instead?

Reply:
No. `0xEFBBBF` will change the value:
![image](https://user-images.githubusercontent.com/5399861/87413662-2b2d2c80-c5fd-11ea-81b9-d363247f02e6.png)

Reply:
Yes, this is the wrong BOM. It may help your case but is going to break others - even adding the BOM might break things. I think you should tell Excel that you need _UTF-16_ when reading the file. That's what you're 'fixing'

Reply:
We meet the same problem in our project, and we use `0xEFBBBF` as default BOM for UTF-8, it will change the value if we use `0xFEFF`. Besides, we have meet other problems, such as the commas were used incorrectly or quotation marks were not displayed properly. If we fix these problems, it will cause other users can not read these files with other tools.

Hm, what I trying to say is, maybe it is not a good idea to change CsvOutputWriter to fit Excel format. It can be done in project before use downloads the csv files or just use Excel to import.

Reply:
@stczwd What tool will change the value if we use `0xFEFF`?

Reply:
Excel. It will change the actual value if we add `0xFEFF` in the front.

Reply:
This change is still wrong. I cannot see merging this.